### PR TITLE
nixos/xmrig: init

### PR DESCRIPTION
**Pull request by ratsclub**
###### Motivation for this change
Added a service to the xmrig miner.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ratsclub — COMMENTED** [review]
I added a comment responding something you mentioned on the other PR.

You also said this:

> Hi there. :wave: 
> 
> I left a review on your module. I hope it helps! If you have any questions please don't hesitate to ask for clarification, help, etc...
> 
> Additional to the review I left I have some concerns about the pass option which takes a secret. Users should not be storing secrets in the nix store, which this module seems to encourage. Did you have any thoughts on how to work around that? I had a few, if you did not...
> 
> Let me know! Thanks rocket

I have nothing in mind about it. May you give me some ideas?

**Review by aanderse — CHANGES_REQUESTED** [review]
Looks good. Only a few very minor issues left. Good job :+1:

**Review by aanderse — APPROVED** [review]
The manual builds, right? You checked?

Are we good to merge? LGTM :+1:

**Comment by ratsclub** [comment]
The conflicts have been resolved! :wink:

File: nixos/modules/services/misc/xmrig.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        ExecStartPre = "${cfg.package}/bin/xmrig --config=${configFile} --dry-run";
+        ExecStart = "${cfg.package}/bin/xmrig --config=${configFile}";
+        DynamicUser = true;
+      };
+    };
+  };
+
+  meta = with lib; {
+    description = "XMRig Mining Software Service";
+    license = licenses.gpl3;
```
Comment (aanderse) [inline nixos/modules/services/misc/xmrig.nix:68]: `gpl3Plus` or `gpl3Only`?

**Review by aanderse — CHANGES_REQUESTED** [review]
I'm so sorry I took too long on this now there is a merge conflict! Please rebase and I promise I'll try to merge this ASAP!

**Comment by ratsclub** [comment]
@aanderse There you go! Thanks for your service and don't worry about the time it took!

**Comment by aanderse** [comment]
Thanks again @ratsclub!

**Pull request merged**